### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by morethanreal**
@cdata PTAL cc @notwaldorf

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<!doctype html>
+<!--
+  @license
+  Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+  This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+  The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+  The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+  Code distributed by Google as part of the polymer project is also
+  subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
```
Comment (cdata) [inline demo/index.html]: Indentation

File: paper-dialog.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../paper-dialog-behavior/paper-dialog-behavior.html">
+<link rel="import" href="../paper-styles/paper-styles.html">
+
+<style is="x-style">
+
+  * {
```
Comment (cdata) [inline paper-dialog.html]: @sjmiles asked us to use `:root` for this now.

**Comment by cdata** [comment]
Minor details, otherwise :+1: 

Many repos. Such redundant.

File: demo/index.html (hunk 1 above)
Comment (morethanreal) [inline demo/index.html]: Fixed.

File: paper-dialog.html (hunk 2 above)
Comment (morethanreal) [inline paper-dialog.html]: Right, and `<style is="custom-style">` too. Fixed.

**Pull request merged**